src/components/Permalink/Permalink.js

```diff
@@ -17,24 +14,119 @@ const propTypes = {
   history: PropTypes.shape({
     replace: PropTypes.func,
   }),
+
+  /**
+   * Layers provider.
+   */
+  layerService: PropTypes.instanceOf(LayerService),
+
+  /**
+   * An ol map.
+   */
+  map: PropTypes.instanceOf(OLMap),
+
+  /**
+   * Params to be written in url.
+   */
+  params: PropTypes.object,
 };
 
 const defaultProps = {
   history: null,
+  layerService: null,
+  map: null,
+  params: {},
 };
 
 /**
  * This component handles permalink logic.
  */
 class Permalink extends Component {
-  componentDidUpdate() {
+  constructor(props) {
+    super(props);
+    this.state = {
+      mapParams: null,
+      layersParams: null,
+    };
+  }
+
+  componentDidMount() {
+    const { map, layerService } = this.props;
+    if (map) {
+      map.on('moveend', () => this.onMapMoved());
+    }
+
+    if (layerService) {
+      this.updateLayerService();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { layerService } = this.props;
+
+    if (layerService !== prevProps.layerService) {
+      this.updateLayerService();
+    }
     this.updateHistory();
   }
 
+  onMapMoved() {
+    const { map } = this.props;
+    const mapView = map.getView();
+    const { center } = mapView.getProperties();
+
+    this.setState({
+      mapParams: {
+        x: center[0],
+        y: center[1],
+        z: mapView.getZoom(),
+      },
+    });
+  }
+
+  updateLayerService() {
+    const { layerService } = this.props;
+    if (layerService) {
+      this.updateLayers(layerService);
+      layerService.on('change:visible', () => this.updateLayers(layerService));
+    }
+  }
+
+  updateLayers(layerService) {
+    this.setState({
+      layersParams: {
+        layers: layerService
+          .getLayers()
```

Comment:
You can use `getLayersAsFlatArray`  it will simplify the rest of the code.

Reply:
Of course! thanks :)